Quick and dirty. Allows me to read xls spreadsheets without having to
save intermediate files.

Index: file.c
===================================================================
RCS file: /home/cvs/elinks/elinks/file.c,v
retrieving revision 1.3
diff -u -r1.3 file.c
--- file.c	2 Jan 2002 23:01:37 -0000	1.3
+++ file.c	15 Jan 2002 23:55:31 -0000
@@ -335,9 +335,24 @@
 		add_to_str(&file, &fl, "</pre></body></html>\n");
 		head = stracpy("\r\nContent-Type: text/html\r\n");
 	} else if (!S_ISREG(stt.st_mode)) {
+		int offset = 0;
+		const int bufsiz=4096;
 		mem_free(name);
+		if (!(file = mem_alloc(bufsiz+1))) {
+			close(h);
+			setcstate(c, S_OUT_OF_MEM); abort_connection(c); return;
+		}
+		while (0 < (r = read(h, file+offset, bufsiz))) {
+			offset += r;
+			if (!(file = mem_realloc(file, offset+bufsiz+1))) {
+				mem_free(file);
+				close(h);
+				setcstate(c, S_OUT_OF_MEM); abort_connection(c); return;
+			}
+		}
+		fl = offset;
+		file[fl] = '\0'; /* NULL-terminate just in case */
 		close(h);
-		setcstate(c, S_FILE_TYPE); abort_connection(c); return;
 	} else {
 		mem_free(name);
 		/* + 1 is there because of bug in Linux. Read returns -EACCES when
